app/models/custom_button.rb

```diff
@@ -133,7 +141,11 @@ def self.parse_uri(uri)
   end
 
   def self.button_classes
-    return BUTTON_CLASSES
+    @button_class_names ||= BUTTON_CLASSES.collect(&:name)
```

Comment:
Class-level caching makes me sad... as this only gets called once per request, I think we can skip it.